content/blog/trustless-lightning-payments-for-content.md

```diff
@@ -0,0 +1,354 @@
+        paymentPoint=payment_point_encoded,
+        iv=initialization_vector,
+        nTime=timestamp,
+        nNonce=nonce,
+        encContent=enc_content,
+    )
+    sig = SignSqueak(private_key, squeak)
+    squeak.SetSignature(sig)
+    return squeak, secret_key
+```
+
+### How a user interacts with a squeak
+
+After a squeak is created, it can be shared and validated on any node.
+
+For example, if Alice authored a squeak, and Bob obtained a copy of the squeak:
+
+* Bob will know that the squeak hash is the same on his node as on any other node (because the hash function is deterministic).
+* Bob will know that the squeak was authored by Alice (because the Alice's pubkey is embedded in the squeak and the squeak hash is signed with Alice's signature).
+* Bob will know that none of the fields were modified after being signed by the Alice (that would change the hash, and the signature would become invalid if that happened).
+* Bob will know that the squeak was created after a certain time (given by the embedded block hash and block height).
+* Bob will know if the squeak is a reply to another squeak (given by the embedded reply hash field).
+* Bob will know that the squeak was encrypted using a scalar that corresponds to a certain
+elliptic curve point. This means that any lightning invoice that matches the payment point
+is guaranteed to decrypt the content upon settlement.
+
+However, if Bob does not have the corresponding secret key:
+
+* Bob will not be able to see the decrypted content of the squeak.
+
+### How a user unlocks the content of a squeak
+
+Now Bob has a choice to make. Is he interested in reading the squeak that Alice authored, given everything he knows about the squeak?
+
+If Bob wants to unlock the content, he can send a message to all of his peers in the network expressing interest in buying the decryption key for the squeak.
+
+Bob's peers in the network (only those who already have a copy of the decryption key) will respond to Bob by sending him invoices as described in the earlier section.
+
+Bob can now browse through the offers, and make a payment to any peer that offered him a valid invoice. Bob knows which invoices are valid because he can validate against `paymentPoint` field of the squeak header, using the elliptic curve math described earlier.
+
+When Bob makes a Lightning payment to one of the sellers, he will obtain the preimage of the invoice. This preimage can be used to get the decryption key:
+
+* The preimage is $s1 + s2$, as described earlier.
+* Bob already knows $s2$, because the seller sent it to him.
+* Bob calculates $s1 = (s1 + s2) - s2$ to obtain the decryption key.
+* Bob can then decrypt the content of the squeak.
+
+Now Bob can read the content of the squeak!
+
+The [GetDecryptedContent](https://github.com/squeaknode/squeak/blob/fde192b8b59b59090c6d2eb39f87d6b41b15c05e/squeak/core/__init__.py#L242-L261) method of the Squeak class looks like this:
+
+```python
+    def GetDecryptedContent(
```

Comment:
This too is code (implementation details) without a real reference in the text. See comment on `MakeSqueak()` above.